bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,329 @@
+          } else if (command instanceof StringType) {
+            logger.trace("Received input command" + command);
+            Input input = new Input(((StringType) command).toString());
+            device.setInput(input);
+          } else {
+            logger.debug("Received unknown channel command" + command);
+          }
+          break;
+        case CHANNEL_AUDIO_MUTE:
+        case CHANNEL_VIDEO_MUTE:
+          boolean isAudioMute = channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE);
+          boolean isVideoMute = channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE);
+          if (isVideoMute || isAudioMute) {
+            if (command == RefreshType.REFRESH) {
+              // refresh both video and audio mute, as it's one request
+              MuteQueryResponseValue muteStatus = device.getMuteStatus();
+              updateState(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE,
+                  muteStatus.isAudioMuted() ? OnOffType.ON : OnOffType.OFF);
+              updateState(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE,
+                  muteStatus.isVideoMuted() ? OnOffType.ON : OnOffType.OFF);
+            } else {
+              if (isAudioMute) {
+                logger.trace("Received audio mute command" + command);
+                boolean muteOn = command == OnOffType.ON;
+                device.setMute(MuteInstructionChannel.AUDIO, muteOn);
+              }
+              if (isVideoMute) {
+                logger.trace("Received video mute command" + command);
+                boolean muteOn = command == OnOffType.ON;
+                device.setMute(MuteInstructionChannel.VIDEO, muteOn);
+              }
+            }
+          }
+      }
+
+      updateStatus(ThingStatus.ONLINE);
+      logger.trace("Successfully handled command {} on channel {}", command, channelUID.getId());
+    } catch (IOException e) {
+      updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+    } catch (ResponseException e) {
+      updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+    } catch (ConfigurationException e) {
+      updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+    } catch (AuthenticationException e) {
+      handleAuthenticationException(e);
+    }
+  }
+
+  @Override
+  public void initialize() {
+    this.setupJob = scheduler.schedule(() -> {
+      try {
+        setupDevice();
+        setupRefreshInterval();
+      } catch (ResponseException | IOException e) {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+      } catch (AuthenticationException e) {
+        handleAuthenticationException(e);
+      } catch (ConfigurationException e) {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+      }
+    }, 0, TimeUnit.SECONDS);
```

Comment:
```suggestion
    });
```

If you start immediately anyway, you can use `execute`. 

Question: what if the initial ommunication fails. Will the binding ever recover from that or stay in communication error forever? As simple solution would be to re-schedule the connection attempt after a given time (except in case of configuration error, this will persist until the user changeds the config)

Reply:
I did not see any binding I use doing that so far. Could you point me to an example that answers the questions arising from that? E.g. which interval to pick, should it be configurable etc...

Reply:
knx (https://github.com/openhab/openhab2-addons/blob/master/bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/client/AbstractKNXClient.java), onewire (https://github.com/openhab/openhab2-addons/blob/31b9b488adc6b2e2e791e73ba7a11314af1117f3/bundles/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwserverBridgeHandler.java#L357)

Reply:
Thanks, a new autoReconnectPeriod parameter has been introduced.